### PR TITLE
Closes #56 eg program

### LINKED ISSUE #56: no source/original program for `admiral_eg.rda`
decision: keep `admiral_eg.rda` for now and create our own version for the 2nd release

### PR DESCRIPTION
**Pull request by vbshuliar**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:10]: Same comment as for `studyid`

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:11]: Same comment as for `studyid`

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:13]: Same comment as for `studyid`

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:109]: Do you even need a function here? What if you pass `USUBJID = unique(vs$USUBJID)` to do all patients at once?

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:119]: Add a comment explaining why you are removing these, it's mysterious otherwise

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:142]: Having these nested `ifelse` is very difficult to read. Your `EGSTRESN` code is equivalent to:
```
EGSTRESN = case_when(
  EGTESTCD == "RR" & EGELTM == "PT5M" ~ floor(rnorm(n(), 543.9985, 80)),
  EGTESTCD == "RR" & EGELTM == "PT3M" ~ floor(rnorm(n(), 536.0161, 80)),
  EGTESTCD == "RR" ~ floor(rnorm(n(), 532.3233, 80)),
  
  EGTESTCD == "HR" & EGELTM == "PT5M" ~ floor(rnorm(n(), 70.04389, 8)),
  EGTESTCD == "HR" & EGELTM == "PT3M" ~ floor(rnorm(n(), 74.27798, 8)),
  EGTESTCD == "HR" ~ floor(rnorm(n(), 74.77461, 8)),
  
  EGTESTCD == "QT" & EGELTM == "PT5M" ~ floor(rnorm(n(), 450.9781, 60)),
  EGTESTCD == "QT" & EGELTM == "PT3M" ~ floor(rnorm(n(), 457.7265, 60)),
  EGTESTCD == "QT" ~ floor(rnorm(n(), 455.3394, 60))
)
```

Can you please update this and other variables accordingly?

File: data-raw/eg.R
Comment (manciniedoardo) [inline data-raw/eg.R:175]: Swap out these nested `ifelse`'s for `case_when`

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @vbshuliar - you did a super job for first pass at this 😄 

I've left some syntax/code comments, then @kaz462 and I can do a second review once these are addressed and look at the actual data as well!

**Review by manciniedoardo — APPROVED** [review]

**Comment by manciniedoardo** [comment]
Thanks @vbshuliar this is super 🚀 

@kaz462 would you be able to give a second review here, including passing this version of `eg` into the admiral ADEG template to see if anything goes wrong? Then maybe we can get the NEST team to take a final look before we merge in.

**Comment by vbshuliar** [comment]
Thank you very much for the opportunity to contribute to pharmaversesdtm and approving the pull request👌

**Comment by manciniedoardo** [comment]
@vbshuliar no problem. Actually can you make one last commit and add yourself to the DESCRIPTION file under authors?

**Comment by manciniedoardo** [comment]
@vbshuliar btw if you have an ORCID you can also add that to the description file.

File: data-raw/eg.R
Comment (kaz462) [inline data-raw/eg.R:165]:
```suggestion
    EGTESTCD = "ECG Test Short Name",
    EGTEST = "ECG Test Name",
```

File: data-raw/eg.R
Comment (kaz462) [inline data-raw/eg.R:170]:
```suggestion
    EGSTRESC = "Character Result/Finding in Std Format",
    EGSTRESN = "Numeric Result/Finding in Standard Units",
```

File: data-raw/eg.R
Comment (kaz462) [inline data-raw/eg.R:166]:
```suggestion
    EGORRES = "Result or Finding in Original Units",
```

File: data-raw/eg.R
Comment (kaz462) [inline data-raw/eg.R:23]:
```suggestion
  EGTESTCD = c("QT", "HR", "RR", "ECGINT"),
```

File: data-raw/eg.R
Comment (kaz462) [inline data-raw/eg.R:41]:
```suggestion
  ), stringsAsFactors = FALSE
```

**Review by kaz462 — COMMENTED** [review]
Thanks @vbshuliar for creating the EG data, looks awesome!

**Comment by kaz462** [comment]
> 
> @kaz462 would you be able to give a second review here, including passing this version of `eg` into the admiral ADEG template to see if anything goes wrong? Then maybe we can get the NEST team to take a final look before we merge in.

I passed this version of `eg` into the admiral `ADEG` template, everything is fine.

**Comment by manciniedoardo** [comment]
@vbshuliar can you work on this next week?

Additionally, if you merge `main` into your issue branch, it should fix the failing Roxygen check 😄

**Pull request merged**
